apps/www/registry/new-york/ui/stepper.tsx

```diff
@@ -0,0 +1,1021 @@
+      children,
+      index,
+      isCompletedStep,
+      isCurrentStep,
+      label,
+      description,
+      icon,
+      hasVisited,
+      state,
+      checkIcon: checkIconProp,
+      errorIcon: errorIconProp,
+      onClickStep,
+    } = props
+
+    const {
+      checkIcon: checkIconContext,
+      errorIcon: errorIconContext,
+      isError,
+      isLoading,
+      variant,
+      onClickStep: onClickStepGeneral,
+      clickable,
+      expandVerticalSteps,
+      styles,
+      scrollTracking,
+      orientation,
+      steps,
+      setStep,
+      isLastStep: isLastStepCurrentStep,
+      previousActiveStep,
+    } = useStepper()
+
+    const opacity = hasVisited ? 1 : 0.8
+    const localIsLoading = isLoading || state === "loading"
+    const localIsError = isError || state === "error"
+
+    const isLastStep = index === steps.length - 1
+
+    const active =
+      variant === "line" ? isCompletedStep || isCurrentStep : isCompletedStep
+    const checkIcon = checkIconProp || checkIconContext
+    const errorIcon = errorIconProp || errorIconContext
+
+    const renderChildren = () => {
+      if (!expandVerticalSteps) {
+        return (
+          <Collapsible open={isCurrentStep}>
+            <CollapsibleContent
+              ref={(node) => {
+                if (
+                  // If the step is the first step and the previous step
+                  // was the last step or if the step is not the first step
+                  // This prevents initial scrolling when the stepper
+                  // is located anywhere other than the top of the view.
+                  scrollTracking &&
+                  ((index === 0 &&
+                    previousActiveStep &&
+                    previousActiveStep === steps.length) ||
+                    (index && index > 0))
+                ) {
+                  node?.scrollIntoView({
+                    behavior: "smooth",
+                    block: "center",
+                  })
+                }
+              }}
+              className="overflow-hidden data-[state=open]:animate-collapsible-down data-[state=closed]:animate-collapsible-up"
+            >
+              {children}
+            </CollapsibleContent>
+          </Collapsible>
+        )
+      }
+      return children
+    }
+
+    return (
+      <div
+        ref={ref}
+        className={cn(
+          "stepper__vertical-step",
+          verticalStepVariants({
+            variant: variant?.includes("circle") ? "circle" : "line",
+          }),
+          isLastStepCurrentStep && "gap-[var(--step-gap)]",
+          styles?.["vertical-step"]
+        )}
+        data-optional={steps[index || 0]?.optional}
+        data-completed={isCompletedStep}
+        data-active={active}
+        data-clickable={clickable || !!onClickStep}
+        data-invalid={localIsError}
+        onClick={() =>
+          onClickStep?.(index || 0, setStep) ||
+          onClickStepGeneral?.(index || 0, setStep)
+        }
```

Comment:
`onClickStep` has a `void` return type, so this OR will always resolve. If that's what we want, I'd suggest making it explicit:

```suggestion
        onClick={() => {
          onClickStep?.(index || 0, setStep);
          onClickStepGeneral?.(index || 0, setStep);
        }}
```

Otherwsise:

```suggestion
        onClick={() => {
          if (onClickStep) {
            onClickStep(index || 0, setStep);
          } else {
            onClickStepGeneral?.(index || 0, setStep);
          }
        }}
```